crates/wasmtime/src/store.rs

```diff
@@ -162,10 +165,32 @@ impl Store {
                 current_poll_cx: Cell::new(ptr::null_mut()),
                 out_of_gas_behavior: Cell::new(OutOfGas::Trap),
                 context_values: RefCell::new(HashMap::new()),
+                limiter: RefCell::new(None),
             }),
         }
     }
 
+    /// Creates a new store to be associated with the given [`Engine`] and using the supplied
+    /// resource limiter.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use wasmtime::{Engine, Store, StoreLimitsBuilder};
+    /// let engine = Engine::default();
+    /// let store = Store::new_with_limits(&engine, StoreLimitsBuilder::new().instances(10).build());
+    /// ```
+    pub fn new_with_limits(engine: &Engine, limiter: impl ResourceLimiter + 'static) -> Self {
+        let store = Store::new(engine);
+
+        {
+            let mut l = store.inner.limiter.borrow_mut();
+            *l = Some(Arc::new(ResourceLimiterProxy::new(&store, limiter)));
```

Comment:
`Arc` is used here but I think `Rc` may suffice? (I didn't see any location that seemed to require thread-safety here)